src/app/header/header.component.ts

```diff
@@ -1,10 +1,17 @@
-import { Component } from '@angular/core';
+import { Component, Input } from '@angular/core';
+import { AuthenticationService } from '../shared/services/authentication.service';
 
 @Component({
   selector: 'app-header',
   templateUrl: './header.component.html',
   styleUrls: ['./header.component.css']
 })
 export class HeaderComponent {
-  
+  constructor(private authService: AuthenticationService) {}
+
+  @Input() userName: string = '';
+
+  onLogOut(): void {
+    this.authService.logout();
```

Comment:
The problem with not page not changes is connected with Angular Routing - this topic will be learned in the future
Yet, temporarily this problem can be solved with the help of `window.location.reload();`

Reply:
Wow. it works! Thank you)